tests/test_package_manager/test_v1.py

```diff
@@ -0,0 +1,234 @@
+
+        original_hash = packages_v1["dev"][EXAMPLE_PACKAGE_ID.to_uri_path]
+        packages_v1["dev"][EXAMPLE_PACKAGE_ID.to_uri_path] = DUMMY_PACKAGE_HASH
+
+        packages_json_file = self.packages_dir_path / "packages.json"
+        packages_json_file.write_text(json.dumps(obj=packages_v1))
+        pm = PackageManagerV1.from_dir(self.packages_dir_path)
+
+        pm.update_package_hashes().dump()
+
+        packages_v1_updated = json.loads(packages_json_file.read_text(encoding="utf-8"))
+
+        assert pm.dev_packages[EXAMPLE_PACKAGE_ID] == original_hash
+        assert (
+            packages_v1_updated["dev"][EXAMPLE_PACKAGE_ID.to_uri_path] == original_hash
+        )
+
+
+class TestHashUpdateThirdParty(BaseAEATestCase):
+    """Test hash update."""
+
+    use_packages_dir: bool = True
+
+    def test_update_package_hashes(self, caplog) -> None:
+        """Test update package hash method."""
+
+        packages_v1 = json.loads(PACKAGE_JSON_FILE.read_text(encoding="utf-8"))
+        original_hash = packages_v1["dev"].pop(EXAMPLE_PACKAGE_ID.to_uri_path)
+        packages_v1["third_party"][EXAMPLE_PACKAGE_ID.to_uri_path] = DUMMY_PACKAGE_HASH
+
+        packages_json_file = self.packages_dir_path / "packages.json"
+        packages_json_file.write_text(json.dumps(obj=packages_v1))
+        pm = PackageManagerV1.from_dir(self.packages_dir_path)
+
+        with caplog.at_level(logging.WARNING):
+            pm.update_package_hashes().dump()
+            packages_v1_updated = json.loads(
+                packages_json_file.read_text(encoding="utf-8")
+            )
+
+            assert pm.third_party_packages[EXAMPLE_PACKAGE_ID] == original_hash
+            assert (
+                packages_v1_updated["third_party"][EXAMPLE_PACKAGE_ID.to_uri_path]
+                == original_hash
+            )
+            assert "Hash change detected for third party package" in caplog.text
+
+
+class TestVerifyFailure(BaseAEATestCase):
+    """Test verify method."""
+
+    use_packages_dir: bool = True
+
+    def test_verify_method(self, caplog) -> None:
+        """Test update package hash method."""
+
+        pm = PackageManagerV1.from_dir(self.packages_dir_path)
+        assert pm.verify() == 0
+
+        packages_v1 = json.loads(PACKAGE_JSON_FILE.read_text(encoding="utf-8"))
+        packages_json_file = self.packages_dir_path / "packages.json"
+
+        packages_v1["dev"][EXAMPLE_PACKAGE_ID.to_uri_path] = DUMMY_PACKAGE_HASH
+        packages_json_file.write_text(json.dumps(obj=packages_v1))
+
+        with caplog.at_level(logging.ERROR):
+            pm = PackageManagerV1.from_dir(self.packages_dir_path)
+
+            assert pm.verify() == 1
+            assert f"Hash does not match for {EXAMPLE_PACKAGE_ID}" in caplog.text
+            assert (
+                f"Dependency check failed; Hash does not match for {EXAMPLE_PACKAGE_ID}"
+                in caplog.text
+            )
```

Comment:
I will not review this as it is copy-pasted and I'm not gonna play a game of spot-the-differences
https://github.com/valory-xyz/open-aea/blob/46cbcc4e6eebc15947e418e75e1d493c3c3e8eac/tests/test_package_manager/test_v0.py#L113-L188